src/libraries/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs

```diff
@@ -190,21 +190,28 @@ public partial class EventSource : System.IDisposable
         protected void WriteEvent(int eventId, long arg1, long arg2) { }
         protected void WriteEvent(int eventId, long arg1, long arg2, long arg3) { }
         protected void WriteEvent(int eventId, long arg1, string? arg2) { }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("EventSource WriteEvent will serialize the whole object graph. Trimmer will not safely handle this case because properties may be trimmed. This can be suppressed if the object is a primitive type")]
         protected void WriteEvent(int eventId, params object?[] args) { }
         protected void WriteEvent(int eventId, string? arg1) { }
         protected void WriteEvent(int eventId, string? arg1, int arg2) { }
         protected void WriteEvent(int eventId, string? arg1, int arg2, int arg3) { }
         protected void WriteEvent(int eventId, string? arg1, long arg2) { }
         protected void WriteEvent(int eventId, string? arg1, string? arg2) { }
         protected void WriteEvent(int eventId, string? arg1, string? arg2, string? arg3) { }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("EventSource WriteEvent will serialize the whole object graph. Trimmer will not safely handle this case because properties may be trimmed. This can be suppressed if the object is a primitive type")]
         [System.CLSCompliantAttribute(false)]
         protected unsafe void WriteEventCore(int eventId, int eventDataCount, System.Diagnostics.Tracing.EventSource.EventData* data) { }
         protected void WriteEventWithRelatedActivityId(int eventId, System.Guid relatedActivityId, params object?[] args) { }
```

Comment:
Does this one need the attribute? It has the same object[] args that WriteEvent has.

Reply:
Yes, due to the same reason as above